### PR TITLE
Port to Python3

### LINKED ISSUE #13: Port to Python3
Unable to run in latest sugar environment.

### PR DESCRIPTION
**Pull request by ghost**
Port to Python3
Displays Useful Message when camera is not Found.
This PR closes https://github.com/sugarlabs/panorama/issues/13

### CODE REVIEW

**Comment by Ovalelephant35** [comment]
Kindly review @quozl @walterbender @chimosky

File: activity.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import os
 import gi
 gi.require_version('Gtk', '3.0')
-from gi.repository import Gtk
+from gi.repository import Gtk , GObject
```
Comment (quozl) [inline activity.py:28]: Changed line does not pass flake8, E203.

File: pano.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
     from pygame.locals import *
     from pygame import camera
 except ImportError:
-    print 'Error in import Pygame. This activity requires Pygame 1.9'
+    print ('Error in import Pygame. This activity requires Pygame 1.9')
```
Comment (quozl) [inline pano.py:36]: Does not require 1.9, the message should either omit the version, or just don't handle the ImportError.

Your edit also triggered flake8 E211.

File: activity.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
```
Comment (quozl) [inline activity.py:1]: The `#!` is not required, remove it instead of changing.

File: pano.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
```
Comment (quozl) [inline pano.py:1]: The `#!` is not required.

File: pano.py [hunk 5]
```diff
@@ -81,7 +81,10 @@ def run(self):
         pygame.camera.init()
         self.fuente = pygame.font.Font(None, 60)
         self.camlist = camera.list_cameras()
-        self.camera = camera.Camera(self.camlist[0], self.size, "RGB")
+        if self.camlist:
+            self.camera = camera.Camera(self.camlist[0], self.size, "RGB")
+        else:
+            print("No cameras detected.")
```
Comment (quozl) [inline pano.py:87]: A child isn't going to see this message.  So it is not useful.  What do you intend to happen if there is no camera?

File: stitcher.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
```
Comment (quozl) [inline stitcher.py:1]: The `#!` is not required.

File: setup.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (quozl) [inline setup.py:1]: This `#!` is required.  Just a reminder not to delete it.

File: sugargame/canvas.py [hunk 8]
```diff
@@ -4,7 +4,7 @@
 from gi.repository import GLib
 from sugar3.activity.activity import PREVIEW_SIZE
 import pygame
-import event
+import sugargame.event as event
```
Comment (quozl) [inline sugargame/canvas.py:7]: Would prefer an upgrade of Sugargame (the code is over in the Sugargame repository) to be in a separate commit.  Also, I see you have not updated the Sugargame version.  Perhaps you didn't reference the repository.

File: sugargame/event.py [hunk 9]
```diff
@@ -168,10 +169,10 @@ def _keyevent(self, widget, event, type):
             self.__keystate[keycode] = type == pygame.KEYDOWN
             if type == pygame.KEYUP:
                 mod = self._keymods()
-            ukey = unichr(Gdk.keyval_to_unicode(event.keyval))
+            ukey = chr(Gdk.keyval_to_unicode(event.keyval))
             if ukey == '\000':
                 ukey = ''
-            evt = pygame.event.Event(type, key=keycode, unicode=ukey, mod=mod)
+            evt = pygame.event.Event(type, key=keycode, str=ukey, mod=mod)
```
Comment (quozl) [inline sugargame/event.py:175]: Your change here, while important, was in the wrong commit.  Please confine changes into the commits.

**Review by quozl — COMMENTED** [review]
An incomplete review.

File: activity.py (hunk 1 above)
Comment (Ovalelephant35) [reply activity.py:28]: Ok will correct it. Thanks

File: pano.py (hunk 2 above)
Comment (Ovalelephant35) [reply pano.py:36]: What can be done here I am unsure?

File: pano.py (hunk 4 above)
Comment (Ovalelephant35) [reply pano.py:1]: Ok will remove it.

File: pano.py (hunk 5 above)
Comment (Ovalelephant35) [reply pano.py:87]: Then nothing can be done except for seeing camera going out of index, i dont think anything else can be done.

File: stitcher.py (hunk 6 above)
Comment (Ovalelephant35) [reply stitcher.py:1]: Ok will remove it.

File: setup.py (hunk 7 above)
Comment (Ovalelephant35) [reply setup.py:1]: Ok will remove it.

File: sugargame/canvas.py (hunk 8 above)
Comment (Ovalelephant35) [reply sugargame/canvas.py:7]: How reference can be done, I am unsure?

File: sugargame/event.py (hunk 9 above)
Comment (Ovalelephant35) [reply sugargame/event.py:175]: Will change it to include both of the change.

File: activity.py (hunk 3 above)
Comment (Ovalelephant35) [reply activity.py:1]: Ok will remove it.

File: pano.py (hunk 2 above)
Comment (quozl) [reply pano.py:36]: Unfortunately you have gone against our contributor's guide recommendation not to use your _master_ branch, otherwise I could just fix it and push a commit to the pull request in order to show you.

```patch
--- a/pano.py
+++ b/pano.py
@@ -28,12 +28,9 @@ from gi.repository import Gtk
 import time
 import logging
 import stitcher
-try:
-    import pygame
-    from pygame.locals import *
-    from pygame import camera
-except ImportError:
-    print ('Error in import Pygame. This activity requires Pygame 1.9')
+import pygame
+from pygame.locals import *
+from pygame import camera
 
 
 class PanoCapture():
```

Comment (quozl) [reply pano.py:36]: The wildcard import is not your problem for this pull request, but it is something that can be fixed later.  It makes flake8 errors and inhibits some of flake8 feature set.

File: pano.py (hunk 5 above)
Comment (quozl) [reply pano.py:87]: I don' t believe nothing can be done.  Other activities that need a camera do various other things.  Think of the child using the software.  What should they be told, if they have a computer without a camera, and they run this activity?  How should they be told?

**Comment by Ovalelephant35** [comment]
https://github.com/sugarlabs/panorama/pull/11
This is related to Camera Issue.

File: sugargame/canvas.py (hunk 8 above)
Comment (quozl) [reply sugargame/canvas.py:7]: Sorry, I don't understand.  Please try again.

Comment (Ovalelephant35) [reply sugargame/canvas.py:7]: The last part of your first message "Perhaps you didn't refernce the repository" I am unsure how it is done.

**Comment by quozl** [comment]
Indeed, thanks.  I've reviewed and merged it.  Please remember to check for pull requests before making a new one.  Your next step could be to compare your work against the new _master_ branch.

**Comment by Ovalelephant35** [comment]
Yes will do that Thanks.

File: sugargame/canvas.py (hunk 8 above)
Comment (quozl) [reply sugargame/canvas.py:7]: Sugargame is a software library maintained in another repository.  We make changes to Sugargame in that other repository, and then add those changes in one commit to each activity that adopts the new release of Sugargame.  Your commits here have changed Sugargame in this repository, rather than adopt a new release of Sugargame.  This is confusing, as it will potentially result in many different versions of Sugargame across the activity set.

Comment (Ovalelephant35) [reply sugargame/canvas.py:7]: Thankyou ok will do that.
